src/FakeItEasy/Configuration/FakeConfigurationManager.cs

```diff
@@ -61,6 +59,123 @@ public IAnyCallConfigurationWithNoReturnTypeSpecified CallTo(object fakeObject)
+            return new PropertySetterConfiguration<TValue>(setterExpression, this.CreateVoidArgumentValidationConfiguration);
+        }
+
+        private static string GetIndexerName(MethodCallExpression methodCallExpression)
+        {
+            if (methodCallExpression != null &&
+                HasThis(methodCallExpression) &&
+                methodCallExpression.Method.IsSpecialName)
+            {
+                var methodName = methodCallExpression.Method.Name;
+                if (methodName.StartsWith("get_", StringComparison.Ordinal))
+                {
+                    return methodName.Substring(4);
+                }
+            }
+
+            return null;
+        }
+
+        private static bool HasThis(MethodCallExpression methodCallExpression)
+        {
+            return (methodCallExpression.Method.CallingConvention & CallingConventions.HasThis) ==
+                   CallingConventions.HasThis;
+        }
+
+        private static MethodCallExpression BuildSetterFromMemberExpression<TValue>(MemberExpression memberExpression)
+        {
+            PropertyInfo propertyInfo = memberExpression.Member as PropertyInfo;
+            if (propertyInfo == null)
+            {
+                throw new ArgumentException("The expression refers to '" + memberExpression.Member.Name +
+                                            "', which is a field, not a property getter.");
+            }
+
+            var propertySetterInfo = propertyInfo.GetSetMethod(nonPublic: true);
+            if (propertySetterInfo == null)
+            {
+                throw new ArgumentException("The property '" + memberExpression.Member.Name + "' does not have a setter.");
+            }
+
+            var arguments = new[] { BuildArgumentThatMatchesAnything<TValue>() };
+
+            return Expression.Call(memberExpression.Expression, propertySetterInfo, arguments);
+        }
+
+        private static Expression BuildArgumentThatMatchesAnything<TValue>()
+        {
+            Expression<Func<TValue>> value = () => A<TValue>.Ignored;
+            return value.Body;
+        }
+
+        private IVoidArgumentValidationConfiguration CreateVoidArgumentValidationConfiguration(LambdaExpression lambda)
+        {
+            var fake = this.expressionParser.GetFakeManagerCallIsMadeOn(lambda);
+            var rule = this.ruleFactory.Invoke(lambda);
+            fake.AddRuleFirst(rule);
+
+            return this.configurationFactory.CreateConfiguration(fake, rule);
+        }
+
+        private string GetExpressionDescription<TValue>(Expression<Func<TValue>> expression)
+        {
+            var matcher = new ExpressionCallMatcher(
+                expression,
+                ServiceLocator.Current.Resolve<ExpressionArgumentConstraintFactory>(),
+                ServiceLocator.Current.Resolve<MethodInfoManager>(),
+                this.callExpressionParser);
+
+            var expressionDescription = matcher.DescriptionOfMatchingCall;

```

Comment:
Creating an `ExpressionCallMatcher` just to get its description doesn't seem quite right to me. Perhaps the description generation should be taken out of `ExpressionCallMatcher`? (I haven't looked at the code yet, so I'm not sure if it makes sense)